src/pages/workspaces/workspace/JobHistory.js

```diff
@@ -181,15 +187,21 @@ const JobHistory = _.flow(
                         as: 'a',
                         target: '_blank',
                         href: `${firecloudRoot}/#workspaces/${namespace}/${name}/monitor/${submissionId}`
-                      }, [
-                        icon('circle-arrow right'),
-                        div({ style: { marginLeft: '0.5rem' } }, ['View job details'])
-                      ]),
+                      }, [menuIcon('circle-arrow right'), 'View job details']),
+                      isTerminal(status) && workflowStatuses['Failed'] &&
+                      submissionEntity && submissionEntity.entityType.endsWith('_set') && h(MenuButton, {
+                        onClick: () => rerunFailures({
+                          namespace, name, submissionId,
+                          configNamespace: methodConfigurationNamespace, configName: methodConfigurationName,
+                          onDone: () => this.refresh(),
+                          ajax: { Workspaces }
```

Comment:
Passing this means that the ajax flow could get interrupted if this component unmounts, e.g. page transition. Perhaps this is one of the cases where we want to use the non-signal version?

Reply:
Good idea.